File: Amazon/ServerSide/Amazon.Solution/Amazon.Core/Entities/Identity/Address.cs
using System.ComponentModel.DataAnnotations;

namespace Amazon.Core.Entities.Identity
{
	public class Address
	{
		public string Id { get; set; } = Guid.NewGuid().ToString();
		[Required]
		public string Country { get; set; }
		
		[Required]
		public string FirstName { get; set; }
		
		[Required]
		public string LastName { get; set; }
		
		[Required]
		public string PhoneNumber { get; set; }
		
		[Required]
		public string StreetAddress { get; set; }

		[Required]
		public string BuildingName { get; set; }

		[Required]
		public string City { get; set; }
		
		[Required]
		public string District { get; set; }

		[Required]
		public string Governorate { get; set; }
		

		public string NearestLandMark { get; set; }

		public string AppUserId { get; set; }
		public virtual AppUser AppUser { get; set; }
	}
}

File: Amazon/ClientSide/Amazon.ClientSide/src/app/Models/address.ts
export class Address 
{
    constructor(id: string, buildingName: string, city: string, governorate: string, country: string, district: string, firstName: string, lastName: string, nearestLandMark: string, phoneNumber: string, streetAddress: string)
    {
        this.id = id;
        this.buildingName = buildingName;  
        this.city = city; 
        this.country = country; 
        this.district = district
        this.firstName = firstName
        this.governorate = governorate;
        this.lastName = lastName;
        this.nearestLandMark = nearestLandMark;
        this.phoneNumber = phoneNumber;
        this.streetAddress = streetAddress;

    }

    id: string;
    buildingName: string;
    city: string;
    country: string;
    district: string;
    firstName: string;
    governorate: string
    lastName: string;
    nearestLandMark: string;
    phoneNumber: string;
    streetAddress: string;
}